src/components/button/Button.js

```diff
@@ -0,0 +1,44 @@
+import './styles.css';
+
+class Button {
+  constructor({ label = '', onClick = undefined, type = 'default', imageSrc = null, id, width = 'auto' }) {
+    this.is = id;
+    this.label = label;
+    this.onClick = onClick;
+    this.type = type;
+    this.imageSrc = imageSrc;
+    this.width = width;
+  }
+
+  render() {
+    const btn = document.createElement(`button`);
+    const img = document.createElement(`img`);
+    img.src = this.imageSrc;
+    btn.appendChild(img);
+    btn.innerText = this.label;
+
+    if (this.type === 'default') {
+      btn.className = 'button default';
+    }
+    if (this.type === 'hof') {
+      btn.className = 'button hof';
+    }
+    if (this.type === 'ptg') {
+      btn.className = 'button ptg';
+    }
+    if (this.width === 'auto') {
+      btn.style.width = 'auto';
+    }
+    if (this.width === 'full') {
```

Comment:
replace all ifs with btn.style.width = this.width;

when creating new Button you have to give auto or percentage of button width instead of auto, full, half.
Now you have to give: auto or some percentage

Reply:
Ok, this make sense, but this way we are adding a level of abstraction which is more flexible and can be changed any time. I read that giving absolute values this way is not necessary a good thing.